lib/schemaUtils.js

```diff
@@ -137,9 +137,16 @@ function safeSchemaFaker(oldSchema, resolveTo, resolveFor, parameterSourceOption
   var prop, key, resolvedSchema, fakedSchema,
     schemaResolutionCache = _.get(schemaCache, 'schemaResolutionCache', {}),
     schemaFakerCache = _.get(schemaCache, 'schemaFakerCache', {});
+  let concreteUtils = DEFAULT_SCHEMA_UTILS;
+  if (components) {
```

Comment:
maybe this is simpler and cleaner 

```
let concreteUtils = components && components.hasOwnProperty('concreteUtils') ?
      components.concreteUtils :
      DEFAULT_SCHEMA_UTILS;
```

Reply:
Done, this is necessary because there are a lot of tests that do not provide all the parameters in the function, and they are not including the concreteUtils. It is easier to handle it here than to provide all parameters in all the functions that do not provide it in tests.